File: 0-1KnapsackProblem.java
class Solution 
{ 
    //Function to return max value that can be put in knapsack of capacity W.
    static int knapSack(int totalWeights, int weights[], int values[], int n) 
    { 
         // your code here
         
         // Creating A 2D Array For Storing The Results
         int[][] dp = new int[values.length + 1][totalWeights + 1];
         
         for(int i=1; i<dp.length; i++){
             for(int j=1; j<dp[0].length; j++){
                 if(j < weights[i - 1]){
                     dp[i][j] = dp[i - 1][j];
                 }
                 else{
                     dp[i][j] = Math.max(dp[i - 1][j], values[i - 1] + dp[i - 1][j - weights[i - 1]]);
                 }
             }
         }
         
         return dp[dp.length - 1][dp[0].length - 1];
         
    } 
}

File: 0-1KnapsackProblem.js
/**
 * @param {number} W
 * @param {number[]} wt
 * @param {number[]} val
 * @param {number} n
 * @returns {number}
*/

class Solution 
{
    //Function to return max value that can be put in knapsack of capacity W.
    knapSack(W, wt, val, n)
    { 
       // code here
       const dp = [];
       for(let i=0; i<val.length + 1; i++) {
           dp[i] = new Array(W + 1).fill(0);
       }
       for(let i=1; i<dp.length; i++) {
           for(let j=1; j<dp[i].length; j++) {
               if(j >= wt[i - 1]){
                   dp[i][j] = Math.max(dp[i-1][j - wt[i - 1]] + val[i - 1], dp[i - 1][j]);
               }
               else {
                   dp[i][j] = dp[i - 1][j];
               }
           }
       }
       return dp[dp.length - 1][dp[0].length - 1];
    }
}
